### ISSUE TITLE
False positive `Variable is never used` for `r#`-names with usage in `format!` macro 

### ISSUE THREAD

**Issue by neonaot** [intellij-rust/intellij-rust#9452]
## Environment

* **IntelliJ Rust plugin version:** 0.4.181.4916-223-nightly
* **Rust toolchain version:** 1.64.0 (a55dd71d5 2022-09-19) x86_64-apple-darwin
* **IDE name and version:** CLion 2022.3 EAP (CL-223.4884.72)
* **Operating system:** macOS 12.5.1
* **Macro expansion:** enabled
* **Additional experimental features:** org.rust.cargo.features.settings.gutter, org.rust.cargo.emulate.terminal, org.rust.cargo.evaluate.build.scripts, org.rust.macros.proc

## Problem description
Analysing names with raw identifiers does not work:
<img width="643" alt="image" src="https://user-images.githubusercontent.com/49211026/193627923-2f6535cd-d50e-4860-9510-c682212aebc5.png">

Note that `println!("{r#fn}")` is incorrect and plugin shows `Invalid format string` error:
<img width="632" alt="image" src="https://user-images.githubusercontent.com/49211026/193627386-506931fe-c266-4cb9-9241-0e918d947aa3.png">

## Steps to reproduce
```rust
fn main() {
    let r#fn = 566; // Variable `fn` is never used 
    println!("{fn}");
}
```

**Comment by Kobzol** [comment]
To fix this, we would have to resolve the named parameter to see if it resolves to something that is a raw identifier. Or, alternatively, we can always prefix the name with the raw modifier:
```
let r#fn = 1;
let foo = 2;
println!("{fn} {foo}");
//v
format_args!("{fn} {foo}", fn = #fn, foo = #foo);
```
This seems to work and resolved unused warnings for both raw and normal identifiers. @vlad20012 WDYT?
